src/engine/tools/engine-suite/src/shared/executor.py

```diff
@@ -28,27 +28,30 @@ def undo(self) -> Optional[str]:
 class Executor:
     def __init__(self):
         self.tasks = []
+        self.has_error = False
     
     def add(self, task: RecoverableTask):
         self.tasks.append(task)
     
-    def execute(self, dry_run: bool = False):
+    def execute(self, without_debug: bool = False, dry_run: bool = False):
         index = 0
         for task in self.tasks:
             if dry_run:
                 print(f'Will execute {index}: {task.task_info}...')
             else:
-                print(f'Executing {index}: {task.task_info}...')
+                if not without_debug:
+                    print(f'Executing {index}: {task.task_info}...')
                 error = task.execute()
                 if error:
-                    print(f'Error: {error}')
-                    if index > 0:
+                    print(f'Error: {task.task_info} --> {error}')
+                    self.has_error = True
+                    if index > 0 and not without_debug:
```

Comment:
This condition makes that if index is > 0, and without_debug is true the executor wont try to revert the completed tasks. See the comment below for suggestions.